### PR TITLE
Changed DataChart to fix an issue with passing Chart props per property when using areas

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataChart to fix an issue with passing Chart props per property when using areas

#### Where should the reviewer start?

DataChart.js

#### What testing has been done on this PR?

enhanced unit test and Stacked Areas story

#### How should this be manually tested?

Stacked Areas story

#### Do Jest tests follow these best practices?

no updates to test approach

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### Do the grommet docs need to be updated?

no

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

**Comment by jcfilben** [comment]
In the stacked areas story when you click 'usage' or 'bonus' it disappears from the legend

https://user-images.githubusercontent.com/54560994/154148663-ccab4383-99f1-45c0-810f-6da57520f646.mp4

**Comment by jcfilben** [comment]
If we set the opacity like this in chart property: `[{ property: 'usage', opacity: 'strong' }, ... ]` do we want the opacity to stay the same regardless of "focusing" on the other property (in this case "bonus")? In cases where we don't specify the opacity through `property: [{ opacity: ... }]` the opacity for the unselected option would change to be lighter. Notice below "bonus" is selected in the legend and the opacity for "usage" is still strong. Is this the behavior we want?
Screen Shot 2022-02-16 at 10 30 20 AM

**Comment by ericsoderberghp** [comment]
> If we set the opacity like this in chart property: `[{ property: 'usage', opacity: 'strong' }, ... ]` do we want the opacity to stay the same regardless of "focusing" on the other property (in this case "bonus")? In cases where we don't specify the opacity through `property: [{ opacity: ... }]` the opacity for the unselected option would change to be lighter. Notice below "bonus" is selected in the legend and the opacity for "usage" is still strong. Is this the behavior we want?
> Screen Shot 2022-02-16 at 10 30 20 AM

For now, yes. we use opacity to help highlight the active data set. If the user wants to set their own opacity, we don't have another mechanism to show that. This is an existing behavior if you have multiple charts in a DataChart.

**Review by jcfilben — APPROVED** [review]

File: src/js/components/DataChart/DataChart.js [hunk 1]
```diff
@@ -574,6 +574,8 @@ const DataChart = forwardRef(
             return prop
               .map((cProp, j) => {
                 const pProp = cProp.property || cProp;
+                const { property, ...propRest } =
+                  typeof cProp === 'object' ? cProp : {};
```
Comment (MikeKingdom) [inline src/js/components/DataChart/DataChart.js:578]: So would this basically be:
```jsx
const { property: pProp, ...propRest } =
  typeof cProp === 'object' ? cProp : { property: cProp };
```

**Pull request merged**